paper_framework_sources/p2079_system_execution_context.bs

```diff
@@ -299,11 +306,69 @@ Run-time replaceability has the following characteristics:
 * Con: implies going over an ABI, and cannot be optimized at link-time.
 * Con: different implementation may allocate resources for the system scheduler at startup, and then, at the start of main, the implementation is replaced (this is mainly a QOI issue).
```

Comment:
I think we should say that to make it safe requires very strict lifetime control. It has an additional downside that you cannot leave this up to the build system. If I link against onetbb I could get the tbb context, if I have to call something at runtime that cannot happen.
```suggestion
* Con: different implementation may allocate resources for the system scheduler at startup, and then, at the start of main, the implementation is replaced (this is mainly a QOI issue).
* Con: requires strict lifetime and ownership control to be safe, and for the user to do the right thing explicitly.
```

So instead, what? We call manually in main, leading to all the problems of having to ask people to do the right things (which they don't, I couldn't get people to get this right within Meta). Or we call  automatically within libraries that are being linked, but that is identical in its downsides to your .so example.

Reply:
I am not oppose to add that. It's not quite related to the modifications that I made (I mean line-wise) but sure, why not.